From: Thomas Renninger <trenn@suse.de>
Subject: x86 cpufreq: Make trace_power_frequency cpufreq driver independent
Patch-Mainline: no - xen specific
References: none

This patch is a xen fixup and depends on this patch:
x86_cpufreq_make_trace_power_frequency_cpufreq_driver_independent.patch
Either both or none must be in.

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 arch/x86/kernel/process-xen.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: linux-2.6.33-master/arch/x86/kernel/process-xen.c
===================================================================
--- linux-2.6.33-master.orig/arch/x86/kernel/process-xen.c
+++ linux-2.6.33-master/arch/x86/kernel/process-xen.c
@@ -359,7 +359,7 @@ static inline int hlt_use_halt(void)
  */
 void xen_idle(void)
 {
-	trace_power_start(POWER_CSTATE, 1);
+	trace_power_start(POWER_CSTATE, 1, smp_processor_id());
 	current_thread_info()->status &= ~TS_POLLING;
 	/*
 	 * TS_POLLING-cleared state must be visible before we
@@ -425,7 +425,7 @@ EXPORT_SYMBOL_GPL(cpu_idle_wait);
  */
 void mwait_idle_with_hints(unsigned long ax, unsigned long cx)
 {
-	trace_power_start(POWER_CSTATE, (ax>>4)+1);
+	trace_power_start(POWER_CSTATE, (ax>>4)+1, smp_processor_id());
 	if (!need_resched()) {
 		if (cpu_has(&current_cpu_data, X86_FEATURE_CLFLUSH_MONITOR))
 			clflush((void *)&current_thread_info()->flags);
@@ -441,7 +441,7 @@ void mwait_idle_with_hints(unsigned long
 static void mwait_idle(void)
 {
 	if (!need_resched()) {
-		trace_power_start(POWER_CSTATE, 1);
+		trace_power_start(POWER_CSTATE, 1, smp_processor_id());
 		if (cpu_has(&current_cpu_data, X86_FEATURE_CLFLUSH_MONITOR))
 			clflush((void *)&current_thread_info()->flags);
 
@@ -463,7 +463,7 @@ static void mwait_idle(void)
  */
 static void poll_idle(void)
 {
-	trace_power_start(POWER_CSTATE, 0);
+	trace_power_start(POWER_CSTATE, 0, smp_processor_id());
 	local_irq_enable();
 	while (!need_resched())
 		cpu_relax();
